pinot-integration-test-base/src/test/java/org/apache/pinot/integration/tests/ClusterIntegrationTestUtils.java

```diff
@@ -295,7 +295,7 @@ public static void buildSegmentsFromAvro(List<File> avroFiles, TableConfig table
     if (numAvroFiles == 1) {
       buildSegmentFromAvro(avroFiles.get(0), tableConfig, schema, baseSegmentIndex, segmentDir, tarDir);
     } else {
-      ExecutorService executorService = Executors.newFixedThreadPool(numAvroFiles);
```

Comment:
Limit number of threads in case there's a large number of files to upload.

Reply:
nit: Maybe we can use a number proportional to number of cores?